src/Components/Components/src/SupplyParameterFromPersistentComponentStateValueProvider.cs

```diff
@@ -0,0 +1,246 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal sealed class SupplyParameterFromPersistentComponentStateValueProvider(PersistentComponentState state) : ICascadingValueSupplier
+{
+    private static readonly ConcurrentDictionary<(string, string, string), byte[]> _keyCache = new();
```

Comment:
How are the values in this cache evicted ? I'm concerned about concurrency bottleneck and memory leak.
Could this cache have instance per circuit ?

Reply:
Also could this cache leak state across users/tenants ? (security)